app/services/product_filter_service.rb

```diff
@@ -0,0 +1,27 @@
+class ProductFilterService
+  def initialize(params)
+    @params = params
+  end
+
+  def filter_products
+    selected_category = @params[:category]
+    selected_vegetarian = @params[:vegetarian]
+    selected_price_order = @params['price-sorting']
+    selected_min_price = @params[:min_price]
+    selected_max_price = @params[:max_price]
+
+    if selected_category != "All"
```

Comment:
Instead of submitting the value `All` you shouldn't submit any value for `params[:category]` - this is really easy with a select input.